### ISSUE TITLE
Could you provide the training codes

### ISSUE THREAD

**Issue by wu-leaf** [brijeshiitg/Pytorch-implementation-of-SRNet#7]
Hi
Thank you very much for opening source the pytorch version of the SRNet network model code.
But could you provide the training codes?

**Comment by 876arham** [comment]
Can you provide the dataset for cover and stego images, I just need the dataset i cant find it online

**Comment by StruninIhor** [comment]
Hi, @brijeshiitg I am using your model and getting the same output values, which does not depend on the input image (I tried S-Uniward 0.1,0.2-0.9bpp and Hugo like with same values), but values are like `[[ 0.04061105 -0.04050248]]` each time (differences are present just in 11th digit). Could you provide some code, how to use it or how did you train it?
P.S.
```
prediction = output.data.max(1)[1]
print(prediction.numpy())
```

That prints each time ```[0]```

**Comment by brijeshiitg** [comment]
> Can you provide the dataset for cover and stego images, I just need the dataset i cant find it online

Please use Bossbase dataset. It is available at: http://dde.binghamton.edu/download/ImageDB/BOSSbase_1.01.zip and you can download steganography algorithms from: http://dde.binghamton.edu/download/stego_algorithms/ . Use random key setting while making the stego images.

**Comment by brijeshiitg** [comment]
> Hi, @brijeshiitg I am using your model and getting the same output values, which does not depend on the input image (I tried S-Uniward 0.1,0.2-0.9bpp and Hugo like with same values), but values are like `[[ 0.04061105 -0.04050248]]` each time (differences are present just in 11th digit). Could you provide some code, how to use it or how did you train it?
> P.S.
> 
> ```
> prediction = output.data.max(1)[1]
> print(prediction.numpy())
> ```
> 
> That prints each time `[0]`

Use data augmentation: rotation, horizontal flip, and vertical flip while feeding your images then your training should converge.

**Comment by 876arham** [comment]
> > Hi, @brijeshiitg I am using your model and getting the same output values, which does not depend on the input image (I tried S-Uniward 0.1,0.2-0.9bpp and Hugo like with same values), but values are like `[[ 0.04061105 -0.04050248]]` each time (differences are present just in 11th digit). Could you provide some code, how to use it or how did you train it?
> > P.S.
> > ```
> > prediction = output.data.max(1)[1]
> > print(prediction.numpy())
> > ```
> > 
> > 
> > That prints each time `[0]`
> 
> Use data augmentation: rotation, horizontal flip, and vertical flip while feeding your images then your training should converge.



> > Can you provide the dataset for cover and stego images, I just need the dataset i cant find it online
> 
> Please use Bossbase dataset. It is available at: http://dde.binghamton.edu/download/ImageDB/BOSSbase_1.01.zip and you can download steganography algorithms from: http://dde.binghamton.edu/download/stego_algorithms/ . Use random key setting while making the stego images.


https://www.kaggle.com/lijiyu/bossbase
how does this dataset differs from the dataset told by you , I mean do we need to change dataset like this because the dataset i have mentioned contains both stego and cover images but I am getting 50% accuracy on it.

**Comment by Adityashar** [comment]
> > > Hi, @brijeshiitg I am using your model and getting the same output values, which does not depend on the input image (I tried S-Uniward 0.1,0.2-0.9bpp and Hugo like with same values), but values are like `[[ 0.04061105 -0.04050248]]` each time (differences are present just in 11th digit). Could you provide some code, how to use it or how did you train it?
> > > P.S.
> > > ```
> > > prediction = output.data.max(1)[1]
> > > print(prediction.numpy())
> > > ```
> > > 
> > > 
> > > That prints each time `[0]`
> > 
> > 
> > Use data augmentation: rotation, horizontal flip, and vertical flip while feeding your images then your training should converge.
> 
> > > Can you provide the dataset for cover and stego images, I just need the dataset i cant find it online
> > 
> > 
> > Please use Bossbase dataset. It is available at: http://dde.binghamton.edu/download/ImageDB/BOSSbase_1.01.zip and you can download steganography algorithms from: http://dde.binghamton.edu/download/stego_algorithms/ . Use random key setting while making the stego images.
> 
> https://www.kaggle.com/lijiyu/bossbase
> how does this dataset differs from the dataset told by you , I mean do we need to change dataset like this because the dataset i have mentioned contains both stego and cover images but I am getting 50% accuracy on it.

Did your accuracy improve from .50 ?

**Comment by qazzombie** [comment]
> > > Hi, @brijeshiitg I am using your model and getting the same output values, which does not depend on the input image (I tried S-Uniward 0.1,0.2-0.9bpp and Hugo like with same values), but values are like `[[ 0.04061105 -0.04050248]]` each time (differences are present just in 11th digit). Could you provide some code, how to use it or how did you train it?
> > > P.S.
> > > ```
> > > prediction = output.data.max(1)[1]
> > > print(prediction.numpy())
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > That prints each time `[0]`
> > 
> > 
> > Use data augmentation: rotation, horizontal flip, and vertical flip while feeding your images then your training should converge.
> 
> > > Can you provide the dataset for cover and stego images, I just need the dataset i cant find it online
> > 
> > 
> > Please use Bossbase dataset. It is available at: http://dde.binghamton.edu/download/ImageDB/BOSSbase_1.01.zip and you can download steganography algorithms from: http://dde.binghamton.edu/download/stego_algorithms/ . Use random key setting while making the stego images.
> 
> https://www.kaggle.com/lijiyu/bossbase how does this dataset differs from the dataset told by you , I mean do we need to change dataset like this because the dataset i have mentioned contains both stego and cover images but I am getting 50% accuracy on it.

What is steganography in this dataset

> > > Hi, @brijeshiitg I am using your model and getting the same output values, which does not depend on the input image (I tried S-Uniward 0.1,0.2-0.9bpp and Hugo like with same values), but values are like `[[ 0.04061105 -0.04050248]]` each time (differences are present just in 11th digit). Could you provide some code, how to use it or how did you train it?
> > > P.S.
> > > ```
> > > prediction = output.data.max(1)[1]
> > > print(prediction.numpy())
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > That prints each time `[0]`
> > 
> > 
> > Use data augmentation: rotation, horizontal flip, and vertical flip while feeding your images then your training should converge.
> 
> > > Can you provide the dataset for cover and stego images, I just need the dataset i cant find it online
> > 
> > 
> > Please use Bossbase dataset. It is available at: http://dde.binghamton.edu/download/ImageDB/BOSSbase_1.01.zip and you can download steganography algorithms from: http://dde.binghamton.edu/download/stego_algorithms/ . Use random key setting while making the stego images.
> 
> https://www.kaggle.com/lijiyu/bossbase how does this dataset differs from the dataset told by you , I mean do we need to change dataset like this because the dataset i have mentioned contains both stego and cover images but I am getting 50% accuracy on it.

Did  you know which steganography in this dataset https://www.kaggle.com/lijiyu/bossbase

> > > Hi, @brijeshiitg I am using your model and getting the same output values, which does not depend on the input image (I tried S-Uniward 0.1,0.2-0.9bpp and Hugo like with same values), but values are like `[[ 0.04061105 -0.04050248]]` each time (differences are present just in 11th digit). Could you provide some code, how to use it or how did you train it?
> > > P.S.
> > > ```
> > > prediction = output.data.max(1)[1]
> > > print(prediction.numpy())
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > That prints each time `[0]`
> > 
> > 
> > Use data augmentation: rotation, horizontal flip, and vertical flip while feeding your images then your training should converge.
> 
> > > Can you provide the dataset for cover and stego images, I just need the dataset i cant find it online
> > 
> > 
> > Please use Bossbase dataset. It is available at: http://dde.binghamton.edu/download/ImageDB/BOSSbase_1.01.zip and you can download steganography algorithms from: http://dde.binghamton.edu/download/stego_algorithms/ . Use random key setting while making the stego images.
> 
> https://www.kaggle.com/lijiyu/bossbase how does this dataset differs from the dataset told by you , I mean do we need to change dataset like this because the dataset i have mentioned contains both stego and cover images but I am getting 50% accuracy on it.

Did  you know which steganography in this dataset https://www.kaggle.com/lijiyu/bossbase

**Comment by brijeshiitg** [comment]
training code has been uploaded now
